waku/node/v2/rpc/wakurpc.nim

```diff
@@ -21,13 +21,12 @@ proc setupWakuRPC*(wakuProto: WakuProto, rpcsrv: RpcServer) =
      result = WakuSubCodec
 
   # TODO: Implement symkey etc logic
-  rpcsrv.rpc("waku_publish") do(topic: string, message: string) -> bool:
-    let data = cast[seq[byte]](message)
+  rpcsrv.rpc("waku_publish") do(topic: string, message: seq[byte]) -> bool:
     # Assumes someone subscribing on this topic
     #let wakuSub = wakuProto.switch.pubsub
     let wakuSub = cast[WakuSub](wakuProto.switch.pubSub.get())
```

Comment:
This doesn't look right to me (@oskarth ?), looks like something isn't being constructed/initialized correctly, this might be because the standard switch setup might not fit your needs in which case a custom one is probably needed for waku. This usage pattern will most likely lead to some issues...

Reply:
This might just be a left-over from some trials, it can probably be hooked up in a cleaner way